src/modules/data/sound.h

```diff
@@ -18,9 +18,15 @@ typedef enum {
 } ChannelLayout;
 
 typedef struct Sound Sound;
+
+typedef uint32_t (*SoundCallback)(Sound* sound, uint32_t offset, uint32_t count, void* data);
+typedef void (*SoundDestroyCallback)(Sound* sound);
```

Comment:
Another minor thing!  Could these function pointers be declared without the "pointer" part, like this:

```c
typedef uint32_t SoundCallback(Sound* sound, uint32_t offset, uint32_t count, void* data);
typedef void SoundDestroyCallback(Sound* sound);
```

The types in the struct/prototypes just need a star like `SoundCallback* read`

I mention it because I won't be able to make this change later without unknowingly breaking your stuff